### PR TITLE
Add new EXIF metadata properties

### LINKED ISSUE #10: Support for more metadata fields?
First off, thanks for making this project! It's been really helpful for loading and grabbing thumbnails of photos, especially RAW photos!

I'm wondering if there are plans to add more metadata fields in the future? Ones that come to mind are lens info, exif data like Title, Description, etc. If this should be in user-land, maybe just exposing the metadata dict in either the `ImageLoader` or `ImageMetadata` structs?

I'm happy to take a first pass at a PR, but wanted to check if there was a reason they were left out from the project.

### PR DESCRIPTION
**Pull request by mike-engel**
Closes #10

This adds a few other useful, technical EXIF properties:
- exposure compensation
- lens maker
- lens model
- flash mode
- metering mode
- white balance mode

Mappings are from (`exiftool`'s spec](https://www.exiftool.org/TagNames/EXIF.html)

### CODE REVIEW

File: Sources/Carpaccio/ImageMetadata.swift [hunk 1]
```diff
@@ -591,6 +669,149 @@ public enum ImageOrientation: String, Codable {
     }
 }
 
+public enum FlashMode: String, Codable {
```
Comment (mz2) [inline Sources/Carpaccio/ImageMetadata.swift:672]: My only nitpick is really the inconsistent capitalisation for these cases. Any particular significance to the combination of capitalisation applied?

Comment (mike-engel) [reply Sources/Carpaccio/ImageMetadata.swift:672]: Yeah, I went back and forth on that. It was mostly for displaying what the values meant. There's probably a better way to do that, so I'll go with the same kebab-case used everywhere else

Comment (mike-engel) [reply Sources/Carpaccio/ImageMetadata.swift:737]: Switched to kebab-case, and added a `humanReadableFlashMode`. It's english only, so if we want to get rid of it, that's fine too!

File: Sources/Carpaccio/ImageMetadata.swift [hunk 2]
```diff
@@ -402,6 +456,71 @@ public struct ImageMetadata: Codable {
...
+        
+        switch (fm) {
+            case .unknown:
+                return "unknown"
+            case .noFlash:
+                return "No Flash"
+            case .fired:
+                return "Fired"
+            case .firedNotReturned:
+                return "Fired, Return not detected"
```
Comment (mz2) [inline Sources/Carpaccio/ImageMetadata.swift:473]: Thanks! The change you made to separate out the codable vs human readable part makes a lot of sense 👍 

I promise this is the last nitpick I will offer on this: I see the precise capitalisation in this new human readable flash mode still follows the `exiftool` documentation page's exact capitalisation, which is kind of strange combo of sentence cased phrases with comma as the separator. Wonder what you'd feel about `"Fired, return not detected"`, `"Off, did not fire"` etc? I'm not a native English speaker so not sure if this is a thing I'm missing. Also not an especially big deal, let me know what you think.

File: Sources/Carpaccio/ImageMetadata.swift (hunk 1 above)
Comment (mz2) [reply Sources/Carpaccio/ImageMetadata.swift:737]: I actually meant to whinge about the odd combo of sentence cased phrases with a comma separating them (commented on this now over at https://github.com/mz2/Carpaccio/pull/13#discussion_r707659667), but this separation of the encoding + human readable string also makes sense 😄

File: Sources/Carpaccio/ImageMetadata.swift (hunk 2 above)
Comment (mike-engel) [reply Sources/Carpaccio/ImageMetadata.swift:473]: Hey @mz2! I see what you're saying now. English is a bit weird (compared to say, German), in that pretty much any capitalization is valid. The way `exiftool` is capitalizing words is valid, but so is yours. I don't really have a preference either way, so I'll edit it to be like your examples. This is your project after all, so you get the final call!

Comment (mz2) [reply Sources/Carpaccio/ImageMetadata.swift]: Appreciated, thank you! And Finnish is even weirded, I promise 🤣

**Review by mz2 — APPROVED** [review]

**Comment by mz2** [comment]
Huge thank you for contributing this!

(...and apologies for the delay in reviewing, had a busy week)

**Comment by mike-engel** [comment]
Thanks for reviewing and merging @mz2. And no worries about the delay. It's quite fast still for open source 😁

**Pull request merged**